CSharpMath.Editor.Tests/CaretTests.cs

```diff
@@ -176,4 +176,101 @@ public class CaretCanStartAndStopBlinking {
       Assert.Equal(MathKeyboardCaretState.Shown, keyboard.CaretState);
     }
   }
+  public class DefaultPlaceholderSettings {
+    public const string ActiveNucleus = "■";
+    public const string RestingNucleus = "□";
+    public static readonly System.Drawing.Color? ActiveColor = null;
+    public static readonly System.Drawing.Color? RestingColor = null;
+  }
+  [CollectionDefinition(nameof(NonParallelPlaceholderTests), DisableParallelization = true)]
+  public class NonParallelPlaceholderTests { }
+  [Collection(nameof(NonParallelPlaceholderTests))]
+  public class DefaultPlaceholder {
+    [Fact]
+    public void LaTeXSettingsPlaceholderIsNewInstance() {
+      Assert.False(LaTeXSettings.Placeholder == LaTeXSettings.Placeholder);
+      // Double check, also verify that its contents are 'fresh':
+      LaTeXSettings.Placeholder.Nucleus = "x";
+      Assert.Equal(DefaultPlaceholderSettings.RestingNucleus, LaTeXSettings.Placeholder.Nucleus);
+      LaTeXSettings.Placeholder.Color = System.Drawing.Color.Green;
+      Assert.Equal(DefaultPlaceholderSettings.RestingColor, LaTeXSettings.Placeholder.Color);
+    }
+    [Fact]
+    public void DefaultPlaceholderAppearance() {
+      Assert.Null(LaTeXSettings.PlaceholderActiveColor);
+      Assert.Null(LaTeXSettings.PlaceholderRestingColor);
+      Assert.Equal(DefaultPlaceholderSettings.ActiveNucleus, LaTeXSettings.PlaceholderActiveNucleus);
+      Assert.Equal(DefaultPlaceholderSettings.RestingNucleus, LaTeXSettings.PlaceholderRestingNucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, LaTeXSettings.Placeholder.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, LaTeXSettings.Placeholder.Color);
+    }
+  }
+  [Collection(nameof(NonParallelPlaceholderTests))]
+  public class CustomizablePlaceholder : IDisposable {
+    public CustomizablePlaceholder() {
+      LaTeXSettings.PlaceholderActiveNucleus = "😀";
+      LaTeXSettings.PlaceholderRestingNucleus = "😐";
+      LaTeXSettings.PlaceholderActiveColor = System.Drawing.Color.Green;
+      LaTeXSettings.PlaceholderRestingColor = System.Drawing.Color.Blue;
+    }
+    public void Dispose() {
+      LaTeXSettings.PlaceholderActiveNucleus = DefaultPlaceholderSettings.ActiveNucleus;
+      LaTeXSettings.PlaceholderRestingNucleus = DefaultPlaceholderSettings.RestingNucleus;
+      LaTeXSettings.PlaceholderActiveColor = DefaultPlaceholderSettings.ActiveColor;
+      LaTeXSettings.PlaceholderRestingColor = DefaultPlaceholderSettings.RestingColor;
+    }
+    [Fact]
+    public async void CustomizedPlaceholderBlinks() {
```

Comment:
Even though xUnit seems to have a custom SynchronizationContext for every test to make `async void` tests work, seeing `async void` outside of event handlers make my eyes bleed. Is there a reason to do this?

Reply:
O, that was just a stupid mistake. I avoid it in general, I know Exceptions are harder to catch in case of async void and there is no other reason to use async void than for button click events etc.
In the context of unit tests I somehow lost alertness for this. I will use async Task.